frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -150,69 +178,89 @@ export default ({
       return mt
     },
     messageType (message) {
       let mt = 'message'
-      switch (message.from) {
-        case messageTypes.NOTIFICATION:
+      switch (message.type) {
+        case MESSAGE_TYPES.NOTIFICATION:
           mt += '-notification'
           break
 
-        case messageTypes.INTERACTIVE:
+        case MESSAGE_TYPES.INTERACTIVE:
           mt += '-interactive'
           break
       }
       return mt
     },
     isCurrentUser (fromId) {
-      return this.currentUserAttr.id === fromId
+      return this.currentUserAttr.username === fromId
     },
     who (message) {
-      const fromId = message.from === messageTypes.NOTIFICATION ? fakeEvents[message.id].from : message.from
-      const user = this.isCurrentUser(fromId) ? this.currentUserAttr : this.details.participants[fromId]
-      if (user) {
-        return user.displayName || user.username
-      }
+      const user = this.isCurrentUser(message.from) ? this.currentUserAttr : this.details.participants[message.from]
+      return user.displayName || user.username || message.from
     },
     variant (message) {
-      if (message.hasFailed) {
-        return this.messageVariants.FAILED
+      if (message.pending) {
+        return MESSAGE_VARIANTS.PENDING
+      } else if (message.hasFailed) {
+        return MESSAGE_VARIANTS.FAILED
       } else {
-        return this.isCurrentUser(message.from) ? this.messageVariants.SENT : this.messageVariants.RECEIVED
+        return this.isCurrentUser(message.from) ? MESSAGE_VARIANTS.SENT : MESSAGE_VARIANTS.RECEIVED
       }
     },
-    avatar (isCurrentUser, fromId) {
-      return isCurrentUser ? this.currentUserAttr.picture : this.details.participants[fromId].picture
+    time (strTime) {
+      return new Date(strTime)
+    },
+    avatar (fromId) {
+      if (fromId === MESSAGE_TYPES.NOTIFICATION || fromId === MESSAGE_TYPES.INTERACTIVE) {
+        return this.currentUserAttr.picture
+      }
+      return this.details.participants[fromId].picture
     },
     isSameSender (index) {
       if (!this.messages[index - 1]) { return false }
+      if (this.messages[index].type !== MESSAGE_TYPES.TEXT) { return false }
+      if (this.messages[index].type !== this.messages[index - 1].type) { return false }
+      const timeBetween = new Date(this.messages[index].datetime).getTime() -
+        new Date(this.messages[index - 1].datetime).getTime()
+      if (timeBetween > MINS_MILLIS * 10) { return false }
       return this.messages[index].from === this.messages[index - 1].from
     },
-
     updateSendAreaHeight (height) {
       this.ephemeral.bodyPaddingBottom = height
     },
-    handleSendMessage (message, replyingMessage = false) {
+    handleSendMessage (message, replyingMessage = null) {
       console.log('sending...')
-      const index = Object.keys(this.messages).length + 1
-      const newMessage = {
-        from: this.currentUserAttr.id,
-        time: new Date(),
+      // Consider only simple TEXT now
```

Comment:
There's a `console.log` above this, it looks like it should be a `console.debug`..?

Reply:
Removed it. No need actually.